### PR TITLE
Fallback check in case OrgManager has fewer permissions. Fixes #240

### LINKED ISSUE #240: OrgManager and SpaceManger should be able to set user roles
A Cloud Foundry user with only an OrgManager or SpaceManager role can run 
`cf set-org-role` and `cf set-space-role` on the the orgs and spaces where they have that role to set permissions for other users. 

However, it's **not** possible to  declare which accounts should have which roles on those orgs and spaces and use the Terraform provider (provisioned with the same account info) to make it so, even though you provide the same information.

Note we are using the [latest version of the provider](https://github.com/cloudfoundry-community/terraform-provider-cf/releases/tag/v0.11.0) and our CF API endpoint is reporting api version 2.146.0.

We've tried doing this two ways First, by specifying the users by name as you would at the command-line, eg
```
resource "cloudfoundry_space_users" "development" {
    space       = "${cloudfoundry_space.development.id}"
    developers  = ["joebob@organization.gov"]
}
```

When we do that, we see
<details>
  <summary>Show output</summary>

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place
Terraform will perform the following actions:
  ~ cloudfoundry_space_users.development
      developers.#:          "0" => "1"
      developers.2835520694: "" => "joebob@organization.gov"
Plan: 0 to add, 1 to change, 0 to destroy.
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.
  Enter a value: yes
cloudfoundry_space_users.development: Modifying... (ID: ff8dfb3c-8802-9eb5-e55a-7c2f4a6c2359)
  developers.#:          "0" => "1"
  developers.2835520694: "" => "joebob@organization.gov"
Error: Error applying plan:
1 error occurred:
	* cloudfoundry_space_users.development: 1 error occurred:
	* cloudfoundry_space_users.development: You are not authorized to perform the requested action
Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```
</details>

We've also tried by specifying it this way:
```
data "cloudfoundry_user" "joebob" {
    name="joebob@organization.gov"
}
resource "cloudfoundry_space_users" "development" {
    space       = "${cloudfoundry_space.development.id}"
    developers  = ["${data.cloudfoundry_user.joebob.id}"]
}
```

When we do that, we see the same as above.

### PR DESCRIPTION
**Pull request by loafoe**
It looks like newer version of the CF API limit
capabilities of OrgManagers. The call to get an 
overview of a user’s orgs could fail. In this case we 
can fall back to a (slower) check by looping through
all current users of the Org. This allows the operation
To proceed. This is crucial when the provider is 
operated by OrgManager accounts as it would otherwise
not allow them to create and populate spaces properly,
even though they can using the CF CLI.
The PR preserves the "fast path" for admin users.

### CODE REVIEW

**Comment by loafoe** [comment]
@ArthurHlt should I target `dev` branch?

**Comment by loafoe** [comment]
@ArthurHlt `data.cloudfoundry_user` has similar issue for OrgManagers and I have a fix that depends on the code in PR that adds an optional `org` field so the guid can be looked up in scope of ORG, which works fine. I can add that to this `PR` or create  new one if you merge, let me know.

**Comment by marcofranssen** [comment]
Seems I'm suffering the same issue.

```bash
Error: Element not found

  on modules/my_module/main.tf line 39, in data "cloudfoundry_org" "org":
  39: data "cloudfoundry_org" "org" {



Error: You are not authorized to perform the requested action

  on modules/my_module/main.tf line 43, in data "cloudfoundry_user" "user":
  43: data "cloudfoundry_user" "user" {
```

My user is a CF admin on the organization.

Would be great if this PR could be merged. Hopefully that resolves also my issue.

**Comment by loafoe** [comment]
@ArthurHlt @janosbinder anything we can do to get this reviewed / merged? Within Philips there are some teams who are blocked on this (and related) issue. We have some capacity to contribute but want to understand your approach. Should we target the `dev` branch?

File: cloudfoundry/utils_clients.go [hunk 1]
```diff
@@ -73,6 +73,22 @@ func UsersToIDs(users []ccv2.User) []interface{} {
 	return ids
 }
 
+func IsErrNotAuthorized(err error) bool {
+	if _, ok := err.(ccerror.ForbiddenError); ok {
+		return true
+	}
+	if httpErr, ok := err.(ccerror.RawHTTPStatusError); ok && httpErr.StatusCode == 403 {
```
Comment (ArthurHlt) [inline cloudfoundry/utils_clients.go:80]: it is a duplicate of the condition below, could you remove one ?

File: .travis.yml [hunk 2]
```diff
@@ -14,6 +14,7 @@ before_install:
   - curl -LSs "https://cli.run.pivotal.io/stable?release=linux64-binary&version=6.43.0" | tar -zvx -C .cf-cli/ -- cf
   - export PATH="$(pwd)/.cf-cli:$PATH"
   - export GO111MODULE=on
+  - go install github.com/golangci/golangci-lint/cmd/golangci-lint
```
Comment (ArthurHlt) [inline .travis.yml:17]: looks like a unwanted addition ?

Comment (loafoe) [reply .travis.yml]: This was to make the build green on `master` , will remove

File: cloudfoundry/utils_clients.go (hunk 1 above)
Comment (loafoe) [reply cloudfoundry/utils_clients.go:80]: ah, this duplication existed as well `IsErrNotFound()` which I used as base, so fixed it there too.

**Comment by loafoe** [comment]
@ArthurHlt addressed both items that you raised 👍

**Comment by ArthurHlt** [comment]
thanks @loafoe

**Pull request merged**